app/scripts/directives/podDonut.js

```diff
@@ -153,7 +154,16 @@ angular.module('openshiftConsole')
             return 'Terminating';
           }
 
+          var warnings = podWarningsFilter(pod);
           if (isTroubledPodFilter(pod)) {
+            // if at least one warning has an "error" severity
+            // return a "Failed" pod phase
+            for (var i = 0; i < warnings.length; i++) {
+              if(warnings[i].severity === "error") {
+                return 'Failed';
+              }
+            }
```

Comment:
This is good JS code, but we use lodash to make working with collections this easier.

```js
var warnings = podWarningsFilter(pod);
if (_.some(warnings, { severity: 'error' }) {
  return 'Failed';
}
```

https://lodash.com/docs/3.10.1